### PR TITLE
ISPN-6357 Deadlock during server start

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-6357

* Fix cluster recovery when a ForkChannel is missing
* Log more info when the cluster recovery fails
* Test startup with ForkChannels

Please cherry-pick to 8.2.x as well.

### CODE REVIEW

**Comment by gustavonalle** [comment]
Testing...

**Comment by gustavonalle** [comment]
@danberindei Looks good, I cannot reproduce the deadlock with this patch :+1:

File: core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java [hunk 1]
```diff
@@ -352,7 +352,9 @@ public void handleClusterView(boolean mergeView, int newViewId) {
                log.tracef("Cluster state recovery interrupted because a member was lost. Will retry.");
          } catch (Exception e) {
             if (clusterManagerStatus.isRunning()) {
-               log.failedToRecoverClusterState(e);
+               CLUSTER.failedToRecoverClusterState(e);
+               eventLogManager.getEventLogger().detail(e)
```
Comment (wburns) [inline core/src/main/java/org/infinispan/topology/ClusterTopologyManagerImpl.java]: :+1:

File: core/src/test/java/org/infinispan/statetransfer/CustomChannelLookup.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.jgroups.JChannel;
+
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * JGroupsChannelLookup implementation that returns an existing channel.
+ *
+ * @author Dan Berindei
+ * @since 8.2
```
Comment (wburns) [inline core/src/test/java/org/infinispan/statetransfer/CustomChannelLookup.java]: 9.0

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated into 8.2.x & master, thanks @danberindei !

**Pull request closed without merging**